core/src/main.py

```diff
@@ -32,9 +32,9 @@
 # as a UI callback for storage, which can be invoked at any time
 import trezor.pin  # noqa: F401
 
-# === Prepare the USB interfaces first. Do not connect to the host yet.
-# usb imports trezor.utils and trezor.io which is a C module
-import usb
```

Comment:
for the next step, please make sure that the original `import usb` statement is kept as it was.

instead, modify `usb.py` to not add interfaces automatically, perhaps create a function `usb.add_interfaces()` which will be called from `main.py` before the second `open()`

Reply:
note that `usb.bus` as well as `usb.iface_*` should remain global, so that code can import them

Reply:
When deleting the `usb_empty` and working just with the one `usb` (state below), I am again getting the Trezor red device with `FATAL ERROR ... core/embed/trezorhal/usb.c:104 ... check_desc_str(dev_info->manufacturer)` after unlocking the device. 

It could be necessary to also recreate the `bus` object in `add_interfaces()` - will try that

```python
# usb imports trezor.utils and trezor.io which is a C module
import usb
# opening empty usb
usb.bus.open("355C817510C0EABF2F147145")  # random 12byte hex

# create an unimport manager that will be reused in the main loop
unimport_manager = utils.unimport()

# unlock the device, unload the boot module afterwards
with unimport_manager:
    import boot
    del boot

# closing the empty USB, adding the interfaces and starting it again
usb.bus.close()
usb.add_interfaces()

import storage.device

usb.bus.open(storage.device.get_device_id())

```

Reply:
Recreating the `bus` helped to get rid of Trezor red screen - and it behaves the same as before - changes in [2e7553d](https://github.com/trezor/trezor-firmware/pull/2129/commits/2e7553d0f1f1c2ffb977596fcf346740c5491421)

Reply:
ah, I see the problem. the instance is not reopenable, per https://github.com/trezor/trezor-firmware/blob/grdddj/scriptpubkey-check-tests-segwit-multisig/core/embed/extmod/modtrezorio/modtrezorio-usb.h#L243-L262

please try to remove the `o->info` clearing (except `serial_number` should be cleared because that is set again by `open()`)

Reply:
Done in 5b9e37b - seems to be fine generally.

There is one difference from the previous state, there are some additional "reset" events going on with the device (output from the `dmesg` command)

```
[236266.736821] usb 1-2: USB disconnect, device number 98
[236267.042219] usb 1-2: new full-speed USB device number 99 using xhci_hcd
[236267.170750] usb 1-2: New USB device found, idVendor=1209, idProduct=53c1, bcdDevice= 2.00
[236267.170756] usb 1-2: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[236267.170760] usb 1-2: Product: TREZOR
[236267.170762] usb 1-2: Manufacturer: SatoshiLabs
[236267.170765] usb 1-2: SerialNumber: 8B70B8E9135C6E17B7322ACC
[236267.177926] hid-generic 0003:1209:53C1.003E: hiddev96,hidraw4: USB HID v1.11 Device [SatoshiLabs TREZOR] on usb-0000:00:14.0-2/input2
[236267.178439] cdc_acm 1-2:1.3: ttyACM0: USB ACM device
[236267.741257] usb 1-2: reset full-speed USB device number 99 using xhci_hcd
[236267.870002] cdc_acm 1-2:1.3: ttyACM0: USB ACM device
[236268.630570] usb 1-2: reset full-speed USB device number 99 using xhci_hcd
[236268.759911] cdc_acm 1-2:1.3: ttyACM0: USB ACM device
```